extensions/notification/utils/config.js

```diff
@@ -0,0 +1,95 @@
+const constants = require('../constants');
+
+/**
+ * Transforms array of configurations to a Map
+ * @param  {Object[]} configs - Array of bucket notification configurations
+ * @return {Map} Map of bucket notification configurations
+ */
+function transform(configs) {
+    const configMap = new Map();
+    configs.forEach(config => {
+        if (config.bucket && config.notificationConfiguration) {
+            const bucket = config.bucket;
+            const conf = config.notificationConfiguration.queueConfigurations;
+            if (configMap.has(bucket)) {
+                const val = configMap.get(bucket);
+                configMap.set(bucket, [...val, ...conf]);
+            } else {
+                configMap.set(bucket, [...conf]);
+            }
+        }
+    });
+    return configMap;
+}
+
+/**
+ * Validate object name against bucket notification configuration filters
+ * @param  {Object[]} filterRules - Filter rules from the bucket notification
+ * configuration.
+ * @param  {string} objectName - Name of the object
+ * @return {boolean} true if filter matches the object name
+ */
+function validateFilter(filterRules, objectName) {
+    return filterRules.some(rule => {
+        const name = rule.name;
+        const value = rule.value;
+        if (name === constants.nameFilter.prefix) {
+            return objectName.startsWith(value);
+        }
+        if (name === constants.nameFilter.suffix) {
+            return objectName.endsWith(value);
+        }
+        return false;
+    });
+}
+
+/**
+ * Validate event against bucket notification configuration event type
+ * @param  {Object[]} bnConfig - Array of bucket specific notification
+ * configurations
+ * @param  {string} event - Type of event
+ * @return {undefined|Object} A bucket notification configuration if found
+ */
+function validateEvent(bnConfig, event) {
+    if (bnConfig.length) {
```

Comment:
If `bnConfig` is always an Array is the check for length necessary? The `0` case would cause `.filter` to return an empty Array which is already handled by the ternary;

Reply:
Ah..yes..forgot to remove this when I refactored.